skyve-war/src/main/java/modules/admin/DataGroup/DataGroupService.java

```diff
@@ -15,12 +14,23 @@
 import org.skyve.persistence.DocumentQuery;
 import org.skyve.persistence.Persistence;
 import org.skyve.util.NullableBeanVisitor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-import jakarta.enterprise.inject.spi.CDI;
-import modules.admin.User.UserService;
+import jakarta.enterprise.inject.Default;
+import jakarta.inject.Inject;
 import modules.admin.domain.DataGroup;
 
-public class DataGroupUtil {
+/**
+ * This class acts as a service layer to encapsulate domain logic.
+ *
+ * Add this line to classes that wish to use it: @Inject private transient DataGroupService dataGroupService;
+ */
+@Default
+public class DataGroupService {
+	private static final Logger LOGGER = LoggerFactory.getLogger(DataGroupService.class);
+	@Inject
+	Persistence pers;
```

Comment:
I think the Skyve injected classes are already safe from serialisation, but we should declare these as `private transient` just in case, and it reinforces the pattern for other developers to copy when injecting classes.